src/components/output/index.tsx

```diff
@@ -10,34 +14,88 @@ type State = {};
 
 export default class App extends Component<Props, State> {
   state: State = {};
-  canvas?: HTMLCanvasElement;
+  canvasLeft?: HTMLCanvasElement;
+  canvasRight?: HTMLCanvasElement;
+  pinchZoomLeft?: PinchZoom;
+  pinchZoomRight?: PinchZoom;
+  retargetedEvents = new WeakSet<Event>();
 
-  constructor() {
-    super();
-  }
-
-  updateCanvas(img: ImageBitmap) {
-    if (!this.canvas) return;
-    const ctx = this.canvas.getContext('2d');
-    if (!ctx) return;
-    ctx.clearRect(0, 0, this.canvas.width, this.canvas.height);
-    ctx.drawImage(img, 0, 0);
+  updateCanvases(img: ImageBitmap) {
+    for (const [i, canvas] of [this.canvasLeft, this.canvasRight].entries()) {
+      if (!canvas) return;
```

Comment:
`continue`? (If there is `canvasRight`, but not `canvasLeft`?)

Reply:
I guess this is one of those "this should never happen" cases.

Should we throw errors in these cases?

Reply:
If it’s never supposed to happen, I’d say throw. Silent return can be frustrating

Reply:
Agreed. I'll fix other places where I've done this.